From 237e516fb6907a4a2ea36ebcfd8751335eafc317 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Fri, 15 Aug 2014 00:54:44 -0400
Subject: [PATCH] Fix corrupted Trapped Chest

CraftBukkit never added Trapped Chests to their "fixTileEntity" code, so trapped chests
that get corrupted crash the server
---
 src/main/java/net/minecraft/server/WorldServer.java | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main/java/net/minecraft/server/WorldServer.java b/src/main/java/net/minecraft/server/WorldServer.java
index 53e62f9..9319345 100644
--- a/src/main/java/net/minecraft/server/WorldServer.java
+++ b/src/main/java/net/minecraft/server/WorldServer.java
@@ -87,7 +87,7 @@ public class WorldServer extends World {
         TileEntity result = super.getTileEntity(i, j, k);
         Block type = getType(i, j, k);
 
-        if (type == Blocks.CHEST) {
+        if (type == Blocks.CHEST || type == Blocks.TRAPPED_CHEST) { // Spigot
             if (!(result instanceof TileEntityChest)) {
                 result = fixTileEntity(i, j, k, type, result);
             }
-- 
1.9.1

